bundles/org.openhab.binding.entsoe/src/main/java/org/openhab/binding/entsoe/internal/client/EntsoeTimeSerie.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.entsoe.internal.client;
+
+import java.math.BigDecimal;
+import java.time.ZonedDateTime;
+
+import javax.measure.Unit;
+import javax.measure.quantity.Energy;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.entsoe.internal.exception.EntsoeResponseException;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.types.State;
+
+/**
+ * @author Jørgen Melhus
+ */
+@NonNullByDefault
+public class EntsoeTimeSerie {
+    private String currency;
+    private Unit<Energy> unit;
+    private Double price;
+    private ZonedDateTime time;
+
+    public EntsoeTimeSerie(String currency, String unit, Double price, ZonedDateTime time)
+            throws EntsoeResponseException {
+        this.currency = currency;
+        this.unit = convertEntsoeUnit(unit);
```

Comment:
For consistency, you could consider moving `calculateDateTime` into this class and letting the constructor here take all those three input parameters. Then all conversions would be in the same place, i.e. `convertEntsoeUnit` and `calculateDateTime`.